packages/replayio/src/utils/authentication/refreshAccessTokenOrThrow.ts

```diff
@@ -0,0 +1,40 @@
+import { maskString } from "../maskString";
+import { authClientId, authHost } from "./config";
+import { debug } from "./debug";
+
+export async function refreshAccessTokenOrThrow(refreshToken: string): Promise<string> {
+  const resp = await fetch(`https://${authHost}/oauth/token`, {
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify({
+      audience: "https://api.replay.io",
+      scope: "openid profile",
+      grant_type: "refresh_token",
+      client_id: authClientId,
+      refresh_token: refreshToken,
+    }),
+  });
+
+  const json: any = await resp.json();
+
+  if (json.error) {
+    debug("OAuth token request failed: %O", json.error);
+
+    throw {
+      id: "auth0-error",
+      message: json.error,
+      refreshToken: maskString(refreshToken),
+    };
```

Comment:
I was afraid to make this change in the old CLI due to lack of tests but it would be really cool to throw actual errors here since we have a fresh start anyway :p

Reply:
Yeah! Agreed.

This was almost the first new part that I wrote, and I wasn't sure if there was a larger reason why we were throwing these weird object values.